### PR TITLE
rpc, cli: describe quality/recency filtering in getnodeaddresses and -addrinfo

### PR DESCRIPTION
**Pull request by jonatack**
Addresses #24278.

```
$ bitcoin-cli help getnodeaddresses
getnodeaddresses ( count "network" )

Return known addresses, after filtering for quality and recency.
These can potentially be used to find new peers in the network.
The total number of addresses known to the node may be higher.
```
```
$ bitcoin-cli -help | grep -A3 addrinfo
  -addrinfo
       Get the number of addresses known to the node, per network and total,
       after filtering for quality and recency. The total number of
       addresses known to the node may be higher.
```

### CODE REVIEW

File: src/bitcoin-cli.cpp [hunk 1]
```diff
@@ -70,7 +70,7 @@ static void SetupCliArgs(ArgsManager& argsman)
...
     argsman.AddArg("-datadir=<dir>", "Specify data directory", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
     argsman.AddArg("-generate", strprintf("Generate blocks immediately, equivalent to RPC getnewaddress followed by RPC generatetoaddress. Optional positional integer arguments are number of blocks to generate (default: %s) and maximum iterations to try (default: %s), equivalent to RPC generatetoaddress nblocks and maxtries arguments. Example: bitcoin-cli -generate 4 1000", DEFAULT_NBLOCKS, DEFAULT_MAX_TRIES), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
-    argsman.AddArg("-addrinfo", "Get the number of addresses known to the node, per network and total.", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
+    argsman.AddArg("-addrinfo", "Get the number of addresses known to the node, per network and total. These are filtered for quality, so the total may be lower than the numbers in the debug log.", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
```
Comment (mzumsande) [inline src/bitcoin-cli.cpp:73]: Not sure if it's good practice to refer to the debug log in RPC helps. How about something like "Get the quality-filtered number of addresses known to the node, per network and total. The total number of addresses known to the node may be higher."
(and something similar below)

**Review by mzumsande — COMMENTED** [review]
Concept ACK, mentioning this distinction makes sense to me.

File: src/bitcoin-cli.cpp (hunk 1 above)
Comment (jonatack) [reply src/bitcoin-cli.cpp:73]: Good point. I'd be tempted to do that, and maybe also mention the debug log as an example so users don't wonder.

Comment (jonatack) [reply src/bitcoin-cli.cpp:73]: Maybe also s/quality/quality and recency/

Comment (jonatack) [reply src/bitcoin-cli.cpp:73]: Thanks, done.

**Review by ghost — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/24370/commits/2ed23881d9e51bfff06fe89ffdb396bd1577bff8

**Review by jarolrod — COMMENTED** [review]
ACK 2ed23881d9e51bfff06fe89ffdb396bd1577bff8
```
./src/bitcoin-cli help getnodeaddresses
getnodeaddresses ( count "network" )

Return known addresses, after filtering for quality and recency.
These can potentially be used to find new peers in the network.
The total number of addresses known to the node may be higher.

...
```

```
./src/bitcoin-cli -help | grep -A3 addrinfo
  -addrinfo
       Get the number of addresses known to the node, per network and total,
       after filtering for quality and recency. The total number of
       addresses known to the node may be higher.
```

**Review by theStack — APPROVED** [review]
ACK 2ed23881d9e51bfff06fe89ffdb396bd1577bff8

**Comment by jonatack** [comment]
Rebased.

**Comment by jonatack** [comment]
Summary: this had 3 ACKs from @prayank23, @jarolrod, and @theStack

**Review by ghost — APPROVED** [review]
reACK https://github.com/bitcoin/bitcoin/pull/24370/commits/ce690847b69eb80b0232f818152dbb1db7c4c61a

**Comment by MarcoFalke** [comment]
@mzumsande Did you want to re-review this before merge, since you left a comment? If not, then it can be merged now.

**Comment by mzumsande** [comment]
>Did you want to re-review this before merge, since you left a comment? If not, then it can be merged now.

Thanks, Code Review ACK ce690847b69eb80b0232f818152dbb1db7c4c61a

**Pull request merged**